### PR TITLE
Replace native-tls with rustls

### PR DESCRIPTION
**Pull request by stjepang**
This is a resurrection of #70, which was reverted in #72 

The issue was that we weren't loading the platform's root certificates, which this PR does load using the `rustls-native-certs` crate.

I have made sure that TLS authentication now really works by running the following program (you need to tweak the path to the `.creds` file before running on your own machine):

```rust
fn main() -> std::io::Result<()> {
    let nc = nats::ConnectionOptions::with_credentials(
        "/Users/stjepan/.nkeys/creds/synadia/stjepang/stjepang.creds",
    )
    .tls_required(true)
    .connect("connect.ngs.global")?;

    let sub = nc.subscribe("hello")?;
    nc.publish("hello", "hi from new-client")?;
    nc.flush()?;

    dbg!(sub.next());
    Ok(())
}
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+  method.
+
+## Breaking Changes
+
+* The underlying TLS implementation has been switched
+  from native-tls to rustls. The previously exported
+  TLS functionality has been removed, and now you can
+  supply certificates with the
+  `Options::add_root_certificate` method.
+
```
Comment (spacejam) [inline CHANGELOG.md:16]: thanks for re-adding this!

File: src/lib.rs [hunk 2]
```diff
@@ -200,18 +195,12 @@ fn inject_io_failure() -> io::Result<()> {
...
 /// Functionality relating to subscribing to a
 /// subject.
 pub mod subscription;
 
 #[doc(hidden)]
-#[deprecated(
-    since = "0.6.0",
-    note = "this has been renamed to `Options`."
-)]
+#[deprecated(since = "0.6.0", note = "this has been renamed to `Options`.")]
```
Comment (spacejam) [inline src/lib.rs:203]:
```suggestion
```

File: tests/connect.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+fn connect_success() {
+    assert!(nats::Options::new().connect("demo.nats.io").is_ok());
+}
+
+#[test]
+fn connect_failure() {
+    assert!(nats::Options::with_credentials("non-existent-file")
+        .connect("demo.nats.io")
+        .is_err());
+}
```
Comment (spacejam) [inline tests/connect.rs:11]:
```suggestion
}

#[test]
fn connect_tls() {
    assert!(nats::Options::new().connect("tls://demo.nats.io:4443").is_ok());
}
```

**Comment by stjepang** [comment]
Looks like there is a spurious CI failure on Windows...

File: src/asynk/connector.rs [hunk 4]
```diff
@@ -27,14 +30,42 @@ pub(crate) struct Connector {
...
+        let mut tls_config = ClientConfig::new();
+
+        // Include platform's native certificate store.
+        match rustls_native_certs::load_native_certs() {
+            Ok(root_store) | Err((Some(root_store), _)) => tls_config
+                .root_store
+                .roots
+                .extend(root_store.roots.into_iter()),
+            Err((None, _)) => {}
+        }
```
Comment (spacejam) [inline src/asynk/connector.rs:50]: :+1:

**Pull request merged**